scio-avro/src/main/scala/com/spotify/scio/avro/AvroDatumFactory.scala

```diff
@@ -64,15 +64,9 @@ private[scio] object GenericRecordDatumFactory extends AvroDatumFactory.GenericD
 private[scio] class SpecificRecordDatumFactory[T <: SpecificRecord](recordType: Class[T])
     extends AvroDatumFactory.SpecificDatumFactory[T](recordType) {
   import SpecificRecordDatumFactory._
-  private class ScioSpecificDatumReader extends SpecificDatumReader[T](recordType) {
-    override def findStringClass(schema: Schema): Class[_] = super.findStringClass(schema) match {
-      case cls if cls == classOf[CharSequence] => classOf[String]
-      case cls                                 => cls
-    }
-  }
 
   override def apply(writer: Schema): DatumWriter[T] = {
-    val datumWriter = new SpecificDatumWriter[T]()
```

Comment:
This constructor uses `SpecificData.get()` instead of expected `SpecificData.getForClass(c)`